### ISSUE TITLE
GDScript icon annotations failing when put in documented location

### ISSUE THREAD

**Issue by Malcolmnixon** [godotengine/godot#71592]
### Godot version

4.0 beta 13

### System information

Windows 11, Vulkan Mobile, NVidia RTX 3070 TI

### Issue description

The online documentation https://docs.godotengine.org/en/latest/tutorials/scripting/gdscript/gdscript_basics.html#example-of-gdscript instructs the user to put the `@icon` annotation after the `class_name` and `extends` line:
![image](https://user-images.githubusercontent.com/1863707/213084377-72f69975-c972-434c-a5ac-0a5e434478fa.png)

This location worked fine in all alpha and beta versions up to beta 12, but as of Godot 4.0 beta 13 putting the `@icon` annotation in this location produces an error:
![image](https://user-images.githubusercontent.com/1863707/213084434-21e1f92f-5921-41c3-8b8a-89b3df27bda4.png)

### Steps to reproduce

In Godot 4.0 beta 13 construct a script with the order recommended in the online documentation and observe the `Annotation "@icon" is not allowed in this level.` error:
```gdscript
class_name Test
@icon("res://addons/godot-xr-tools/editor/icons/body.svg")
```

Swap the order to the following and observe the error goes away:
```gdscript
@icon("res://addons/godot-xr-tools/editor/icons/body.svg")
class_name Test
```

### Minimal reproduction project

N/A

**Comment by Malcolmnixon** [comment]
It looks to be an intentional change with https://github.com/godotengine/godot/pull/67774.

Firstly this change is undocumented and the online documentation should have been updated. 

Secondarily it breaks all existing use of @icons. Godot 4.0 beta-12 forces it in one order, and Godot 4.0 beta-13 forces it in the opposite order. This forces projects and users to simultaneously update their projects as they upgrade to the new Beta. It should probably have had a grace-period to let people migrate their projects and files.

**Comment by adamscott** [comment]
I suggest to close this issue, as this is the wrong forum. The code works as intended, it's the documentation that is out of order.

I opened a issue at godotengine/godot-docs#6629, which is specifically about the documentation.

I created a PR to fix this too godotengine/godot-docs#6628.

> It should probably have had a grace-period to let people migrate their projects and files.

As this is still a beta release, the API is still prone to change, even if we are diligent about this issue. Sorry for the inconvenience.

**Comment by akien-mga** [comment]
Would it be possible to make this error a warning instead, and keep the existing code working? I think that would be a much friendlier way to make users update their code without breaking compat.

@aaronfranke @vnen

**Comment by adamscott** [comment]
> I suggest to close this issue, as this is the wrong forum. The code works as intended, it's the documentation that is out of order.

Sorry, my bad, the class reference for `@icon` must be edited in this repo.

**Comment by aaronfranke** [comment]
It would be possible, but I don't recommend it. I think we should just update the documentation.

**Comment by akien-mga** [comment]
Fixed by #71620 and https://github.com/godotengine/godot-docs/pull/6628.
